flamingo/flamingo.c

```diff
@@ -163,14 +164,12 @@ void flamingo_destroy(flamingo_t* flamingo) {
 
 	// If we didn't inherit our scope stack, free it and all the scopes on it.
 
-	if (!flamingo->inherited_env) {
+	if (!flamingo->inherited_env && flamingo->env != NULL) {
```

Comment:
can `flamingo->env` ever be NULL? seems to me it is either set to the inherited environment, or allocated if not in `flamingo_run`

Reply:
yup indeed :)